File: 125-valid-palindrome/valid-palindrome.js
/**
 * @param {string} s
 * @return {boolean}
 */
var isPalindrome = function(s) {

    s=s.toLowerCase();
    let filteredStr = revStr = '';

    for(let i=0;i<s.length;i++) {

       if(s[i]>='a' && s[i]<='z' || s[i]>='0' && s[i]<='9') {
        filteredStr +=s[i];
        revStr = s[i]+revStr;

       }

    }
    return filteredStr === revStr;

    
};

File: 125-valid-palindrome/valid-palindrome.java
class Solution {
    public boolean isPalindrome(String s) {

        s = s.toLowerCase();
        String filteredStr = "";

        for(int i=0;i<s.length();i++) {
            char c = s.charAt(i);
            if((c>='a' && c<='z') || (c>='0' && c<='9'))  filteredStr += c;
        }

        String reversedStr = "";
        for(int i = filteredStr.length()-1;i>=0;i--) {
            reversedStr += filteredStr.charAt(i);

        }

        return filteredStr.equals(reversedStr);
        
    }
}